Add more docs on permissions in downstream jobs (#851)
packit/packit.dev: docs/fedora-releases-guide/index.md

`allowed_builders` in the [job configuration](/docs/configuration/downstream/bodhi_update).

For a configuration example and retriggering, see [dist-git onboarding](./dist-git-onboarding.md#bodhi-update-job).

> # Permission Handling

Reviewer: ```suggestion
# Permission overview
```

> The downstream automation is a chain of jobs: propose downstream/pull from upstream -> koji build -> bodhi update.

Reviewer: ```suggestion
The downstream automation is a pipeline of jobs: propose_downstream/pull_from_upstream -> koji_build -> bodhi_update.
```

Technically, the first part is optional, one can start a pipeline with a direct push. Should we mention that somehow?

Author: What about something like this:

```suggestion
The downstream automation is a pipeline of jobs: propose_downstream/pull_from_upstream/your-own commits -> koji_build -> bodhi_update.
```

Reviewer: But it could also be Packit commits pushed by a user :slightly_smiling_face: 
Actually, it should look more like this:
`propose_downstream/pull_from_upstream -> Packit PR merge/dist-git push -> koji_build -> bodhi_update.`

Reviewer: Or perhaps this is more accurate:
```
propose_downstream/pull_from_upstream -> Packit PR merge -> koji_build -> bodhi_update
dist-git push -> koji_build -> bodhi_update
```

> This chain can break if Packit does not find the right permissions.

Reviewer 2: ```suggestion
This chain can break (resulting in no followup job) if the permissions are not configured correctly in the Packit configuration.
```

### Propose downstream/Pull from upstream -> Koji build

> A Koji build can be triggered by a merged pull request, a direct push or a comment.

Reviewer: ```suggestion
A Koji build can be triggered by a merged pull request, a direct push or a PR comment.
```

* If you only merge Packit created pull requests, you don't need to specify other settings.

> * If you create your own pull request and you want Packit to automatically trigger a Koji build when you merge it you need to be in the [`allowed_pr_authors`](https://packit.dev/docs/configuration/downstream/koji_build#optional-parameters) list, even when you are reusing Packit commits.

Reviewer: ```suggestion
* If you open your own pull request and you want Packit to automatically trigger a Koji build when you merge it you need to be in the [`allowed_pr_authors`](https://packit.dev/docs/configuration/downstream/koji_build#optional-parameters) list, even in case the PR contains commits created by Packit.
```

> On the other side, if you want to break the chain because, as an example, you want to build the package by yourself in side-tags then you can create a new pull request, re-using the Packit commits, and you have to be sure you are not in the [`allowed_pr_authors`](https://packit.dev/docs/configuration/downstream/koji_build#optional-parameters) list.

Reviewer 2: am I mistaken or this is the use-case that actually doesn't work now 100% because of packit/packit-service#2359 ?

Author: Yes, at the moment we could have problems in both cases (when they want to trigger a build and when they don't) if they create a PR reusing the Packit commits. And unfortunately, the music user yesterday showed us that we also have a problem with direct pushes if the commits are taken from a Packit branch with a related PR.

Reviewer 2: thanks for confirming, then I am thinking what would be the best, whether waiting for merging this PR after the issue is resolved, or merging  now and mentioning the issue until it is fixed

Reviewer: I would probably mention the issue, as it can take some time before the change makes it to the production src.fp.o instance.

> On the other side, if you want to break the chain because, as an example, you want to build the package by yourself in side-tags then you can create a new pull request, re-using the Packit commits, and you have to be sure you are not in the [`allowed_pr_authors`](https://packit.dev/docs/configuration/downstream/koji_build#optional-parameters) list.

Reviewer: ```suggestion
On the other hand, if you want to break the chain because, for example, you want to build the package by yourself in a side-tag then you can create a new pull request containing Packit commits and you have to make sure you are not in the [`allowed_pr_authors`](https://packit.dev/docs/configuration/downstream/koji_build#optional-parameters) list.
```

> * If you push the Packit commits directly into your dist-git project (not merging the Packit pull request) but you still want Packit to react and to trigger a Koji build then you need to set [`allowed_committers`](https://packit.dev/docs/configuration/downstream/koji_build#optional-parameters); you will be the *committer* and Packit the commit's *author*.

Reviewer: ```suggestion
* If you push Packit commits directly to dist-git (not merging Packit pull request) but you still want Packit to react and trigger a Koji build then you need to add yourself to the [`allowed_committers`](https://packit.dev/docs/configuration/downstream/koji_build#optional-parameters) list; Packit is commit *author* but you are the *committer*.
```

> * If you want to trigger a Koji build by comment you need to have `packager` permissions in dist-git.

Reviewer: ```suggestion
* If you want to trigger a Koji build by a PR comment you need be in the `packager` dist-git group.
```

### Koji build -> Bodhi update

> A Bodhi update is triggered by a finished Koji build or a comment.

Reviewer: ```suggestion
A Bodhi update is triggered by a finished Koji build or a PR comment.
```

* If the Koji builds are always submitted by Packit you don't need to specify other settings.

> * If you submit the Koji builds by your own, Packit can still create automatically the Bodhi updates but you need to be in the [`allowed_builders`](https://packit.dev/docs/configuration/downstream/bodhi_update#optional-parameters) list.

Reviewer 2: ```suggestion
* If you submit the Koji builds on your own, Packit can still automatically create the Bodhi updates but you need to be specified in the [`allowed_builders`](https://packit.dev/docs/configuration/downstream/bodhi_update#optional-parameters) list.
```

> * If you want to trigger a Bodhi update by comment you need to have *write permissions* on the project.

Reviewer 2: ```suggestion
* If you want to trigger a Bodhi update by comment you need to have *write permissions* on the project and be a packager.
```

Reviewer 2: iirc

> * If you want to trigger a Bodhi update by comment you need to have *write permissions* on the project and be a `packager`.

Reviewer: ```suggestion
* If you want to trigger a Bodhi update by a PR comment you need to have *write permissions* on the project and be in the `packager` dist-git group.
```

`allowed_builders` in the [job configuration](/docs/configuration/downstream/bodhi_update).

For a configuration example and retriggering, see [dist-git onboarding](./dist-git-onboarding.md#bodhi-update-job).

# Permission overview

> The downstream automation is a pipeline of jobs: propose downstream/pull from upstream -> packit PR merge/dist-git push -> koji build -> bodhi update.

Reviewer: ```suggestion
The downstream automation is a pipeline of jobs:
propose_downstream/pull_from_upstream -> Packit PR merge -> koji_build -> bodhi_update
or:
dist-git push/non-Packit PR merge -> koji_build -> bodhi_update
```
